### ISSUE TITLE
Encountering error when running poppunk assign

### ISSUE THREAD

**Issue by erinyoung** [bacpop/PopPUNK#337]
**Versions**

```
$ conda list
# packages in environment at /home/eriny/miniconda3/envs/poppunk:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
_x86_64-microarch-level   4               2_skylake_avx512    conda-forge
aom                       3.9.1                hac33072_0    conda-forge
at-spi2-atk               2.38.0               h0630a04_3    conda-forge
at-spi2-core              2.40.3               h0630a04_0    conda-forge
atk-1.0                   2.38.0               h04ea711_2    conda-forge
biopython                 1.84            py312h9a8786e_0    conda-forge
brotli                    1.1.0                hb9d3cd8_2    conda-forge
brotli-bin                1.1.0                hb9d3cd8_2    conda-forge
brotli-python             1.1.0           py312h2ec8cdc_2    conda-forge
bzip2                     1.0.8                h4bc722e_7    conda-forge
c-ares                    1.34.2               heb4867d_0    conda-forge
ca-certificates           2024.8.30            hbcca054_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cairo                     1.18.0               hebfffa5_3    conda-forge
cairomm-1.16              1.16.2               h7e731d7_1    conda-forge
certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge
cffi                      1.17.1          py312h06ac9bb_0    conda-forge
charset-normalizer        3.4.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.3.0           py312h68727a3_2    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
dav1d                     1.2.1                hd590300_0    conda-forge
dbus                      1.13.6               h5008d03_3    conda-forge
dendropy                  5.0.1              pyhdfd78af_0    bioconda
docopt                    0.6.2                      py_1    conda-forge
epoxy                     1.5.10               h166bdaf_1    conda-forge
expat                     2.6.3                h5888daf_0    conda-forge
ffmpeg                    7.1.0           gpl_hbb807a5_703    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_3    conda-forge
fontconfig                2.15.0               h7e30c49_1    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.54.1          py312h178313f_1    conda-forge
freetype                  2.12.1               h267a509_2    conda-forge
fribidi                   1.0.10               h36c2ea0_0    conda-forge
gdk-pixbuf                2.42.12              hb9ae30d_0    conda-forge
gettext                   0.22.5               he02047a_3    conda-forge
gettext-tools             0.22.5               he02047a_3    conda-forge
glib-tools                2.82.2               h4833e2c_0    conda-forge
gmp                       6.3.0                hac33072_2    conda-forge
graph-tool                2.79            py312hf67fd38_400    conda-forge
graph-tool-base           2.79            py312hfb2b2c4_400    conda-forge
graphite2                 1.3.13            h59595ed_1003    conda-forge
gtk3                      3.24.43              hd6cba2e_1    conda-forge
h2                        4.1.0              pyhd8ed1ab_0    conda-forge
h5py                      3.12.1          nompi_py312hedeef09_102    conda-forge
harfbuzz                  9.0.0                hda332d3_1    conda-forge
hdbscan                   0.8.39          py312hc0a28a1_0    conda-forge
hdf5                      1.14.3          nompi_hdf9ad27_105    conda-forge
hicolor-icon-theme        0.17                 ha770c72_2    conda-forge
hpack                     4.0.0              pyh9f0ad1d_0    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge
icu                       75.1                 he02047a_0    conda-forge
idna                      3.10               pyhd8ed1ab_0    conda-forge
joblib                    1.4.2              pyhd8ed1ab_0    conda-forge
kernel-headers_linux-64   3.10.0              he073ed8_18    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
kiwisolver                1.4.7           py312h68727a3_0    conda-forge
krb5                      1.21.3               h659f571_0    conda-forge
lame                      3.100             h166bdaf_1003    conda-forge
lcms2                     2.16                 hb7c19ff_0    conda-forge
ld_impl_linux-64          2.43                 h712a8e2_2    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libabseil                 20240722.0      cxx17_h5888daf_1    conda-forge
libaec                    1.1.3                h59595ed_0    conda-forge
libasprintf               0.22.5               he8f35ee_3    conda-forge
libasprintf-devel         0.22.5               he8f35ee_3    conda-forge
libass                    0.17.3               h1dc1e6a_0    conda-forge
libblas                   3.9.0           25_linux64_openblas    conda-forge
libboost                  1.86.0               hb8260a3_2    conda-forge
libboost-python           1.86.0          py312hc39e661_2    conda-forge
libbrotlicommon           1.1.0                hb9d3cd8_2    conda-forge
libbrotlidec              1.1.0                hb9d3cd8_2    conda-forge
libbrotlienc              1.1.0                hb9d3cd8_2    conda-forge
libcblas                  3.9.0           25_linux64_openblas    conda-forge
libcups                   2.3.3                h4637d8d_4    conda-forge
libcurl                   8.10.1               hbbe4b11_0    conda-forge
libdeflate                1.22                 hb9d3cd8_0    conda-forge
libdrm                    2.4.123              hb9d3cd8_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libegl                    1.7.0                ha4b6fd6_1    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libexpat                  2.6.3                h5888daf_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc                    14.2.0               h77fa898_1    conda-forge
libgcc-ng                 14.2.0               h69a702a_1    conda-forge
libgettextpo              0.22.5               he02047a_3    conda-forge
libgettextpo-devel        0.22.5               he02047a_3    conda-forge
libgfortran               14.2.0               h69a702a_1    conda-forge
libgfortran-ng            14.2.0               h69a702a_1    conda-forge
libgfortran5              14.2.0               hd5240d6_1    conda-forge
libgirepository           1.82.0               h0dcfedc_0    conda-forge
libgl                     1.7.0                ha4b6fd6_1    conda-forge
libglib                   2.82.2               h2ff4ddf_0    conda-forge
libglvnd                  1.7.0                ha4b6fd6_1    conda-forge
libglx                    1.7.0                ha4b6fd6_1    conda-forge
libgomp                   14.2.0               h77fa898_1    conda-forge
libhwloc                  2.11.2          default_he43201b_1000    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libjpeg-turbo             3.0.0                hd590300_1    conda-forge
liblapack                 3.9.0           25_linux64_openblas    conda-forge
libnghttp2                1.64.0               h161d5f1_0    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libopenblas               0.3.28          pthreads_h94d23a6_0    conda-forge
libopenvino               2024.4.0             hac27bb2_2    conda-forge
libopenvino-auto-batch-plugin 2024.4.0             h4d9b6c2_2    conda-forge
libopenvino-auto-plugin   2024.4.0             h4d9b6c2_2    conda-forge
libopenvino-hetero-plugin 2024.4.0             h3f63f65_2    conda-forge
libopenvino-intel-cpu-plugin 2024.4.0             hac27bb2_2    conda-forge
libopenvino-intel-gpu-plugin 2024.4.0             hac27bb2_2    conda-forge
libopenvino-intel-npu-plugin 2024.4.0             hac27bb2_2    conda-forge
libopenvino-ir-frontend   2024.4.0             h3f63f65_2    conda-forge
libopenvino-onnx-frontend 2024.4.0             h5c8f2c3_2    conda-forge
libopenvino-paddle-frontend 2024.4.0             h5c8f2c3_2    conda-forge
libopenvino-pytorch-frontend 2024.4.0             h5888daf_2    conda-forge
libopenvino-tensorflow-frontend 2024.4.0             h6481b9d_2    conda-forge
libopenvino-tensorflow-lite-frontend 2024.4.0             h5888daf_2    conda-forge
libopus                   1.3.1                h7f98852_1    conda-forge
libpciaccess              0.18                 hd590300_0    conda-forge
libpng                    1.6.44               hadc24fc_0    conda-forge
libprotobuf               5.28.2               h5b01275_0    conda-forge
librsvg                   2.58.4               hc0ffecb_0    conda-forge
libsqlite                 3.47.0               hadc24fc_1    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx                 14.2.0               hc0a3c3a_1    conda-forge
libstdcxx-ng              14.2.0               h4852527_1    conda-forge
libtiff                   4.7.0                he137b08_1    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libva                     2.22.0               h8a09558_1    conda-forge
libvpx                    1.14.1               hac33072_0    conda-forge
libwebp-base              1.4.0                hd590300_0    conda-forge
libxcb                    1.17.0               h8a09558_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxkbcommon              1.7.0                h2c5496b_1    conda-forge
libxml2                   2.13.4               hb346dea_2    conda-forge
libzlib                   1.3.1                hb9d3cd8_2    conda-forge
mandrake                  1.2.2           py312h4eb2499_10    conda-forge
matplotlib-base           3.9.2           py312hd3ec401_1    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
ncurses                   6.5                  he02047a_1    conda-forge
networkx                  3.4.2              pyhd8ed1ab_1    conda-forge
numpy                     2.1.3           py312h58c1407_0    conda-forge
ocl-icd                   2.3.2                hd590300_1    conda-forge
openblas                  0.3.28          pthreads_h6ec200e_0    conda-forge
openh264                  2.4.1                h59595ed_0    conda-forge
openjpeg                  2.5.2                h488ebb8_0    conda-forge
openssl                   3.3.2                hb9d3cd8_0    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandas                    2.2.3           py312hf9745cd_1    conda-forge
pango                     1.54.0               h4c5309f_1    conda-forge
pcre2                     10.44                hba22ea6_2    conda-forge
pillow                    11.0.0          py312h7b63e92_0    conda-forge
pip                       24.3.1             pyh8b19718_0    conda-forge
pixman                    0.43.2               h59595ed_0    conda-forge
plotly                    5.24.1             pyhd8ed1ab_0    conda-forge
poppunk                   2.7.0           py312h2f68563_0    bioconda
pp-sketchlib              2.1.3           py312he3b2311_3    conda-forge
pthread-stubs             0.4               hb9d3cd8_1002    conda-forge
pugixml                   1.14                 h59595ed_0    conda-forge
pycairo                   1.27.0          py312h51b637d_0    conda-forge
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pygobject                 3.50.0          py312hf4b392c_1    conda-forge
pyparsing                 3.2.0              pyhd8ed1ab_1    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.12.7          hc5c86c4_0_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-tzdata             2024.2             pyhd8ed1ab_0    conda-forge
python_abi                3.12                    5_cp312    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
qhull                     2020.2               h434a139_5    conda-forge
rapidnj                   2.3.2                h4ac6f70_5    bioconda
readline                  8.2                  h8228510_1    conda-forge
requests                  2.32.3             pyhd8ed1ab_0    conda-forge
scikit-learn              1.5.2           py312h7a48858_1    conda-forge
scipy                     1.14.1          py312h62794b6_1    conda-forge
setuptools                75.3.0             pyhd8ed1ab_0    conda-forge
sigcpp-3.0                3.6.0                h59595ed_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.2.1                ha2e4443_0    conda-forge
sparsehash                2.0.4                hcb278e6_1    conda-forge
svt-av1                   2.3.0                h5888daf_0    conda-forge
sysroot_linux-64          2.17                h4a8ded7_18    conda-forge
tbb                       2022.0.0             hceb3a55_0    conda-forge
tenacity                  9.0.0              pyhd8ed1ab_0    conda-forge
threadpoolctl             3.5.0              pyhc1e730c_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tqdm                      4.66.6             pyhd8ed1ab_0    conda-forge
treeswift                 1.1.44             pyh7e72e81_0    bioconda
tzdata                    2024b                hc8b5060_0    conda-forge
unicodedata2              15.1.0          py312h66e93f0_1    conda-forge
urllib3                   2.2.3              pyhd8ed1ab_0    conda-forge
wayland                   1.23.1               h3e06ad9_0    conda-forge
wayland-protocols         1.37                 hd8ed1ab_0    conda-forge
wheel                     0.44.0             pyhd8ed1ab_0    conda-forge
x264                      1!164.3095           h166bdaf_2    conda-forge
x265                      3.5                  h924138e_3    conda-forge
xkeyboard-config          2.43                 hb9d3cd8_0    conda-forge
xorg-libice               1.1.1                hb9d3cd8_1    conda-forge
xorg-libsm                1.2.4                he73a12e_1    conda-forge
xorg-libx11               1.8.10               h4f16b4b_0    conda-forge
xorg-libxau               1.0.11               hb9d3cd8_1    conda-forge
xorg-libxaw               1.0.16               hb9d3cd8_0    conda-forge
xorg-libxcomposite        0.4.6                hb9d3cd8_2    conda-forge
xorg-libxcursor           1.2.2                hb9d3cd8_0    conda-forge
xorg-libxdamage           1.1.6                hb9d3cd8_0    conda-forge
xorg-libxdmcp             1.1.5                hb9d3cd8_0    conda-forge
xorg-libxext              1.3.6                hb9d3cd8_0    conda-forge
xorg-libxfixes            6.0.1                hb9d3cd8_0    conda-forge
xorg-libxi                1.8.2                hb9d3cd8_0    conda-forge
xorg-libxinerama          1.1.5                h5888daf_1    conda-forge
xorg-libxmu               1.2.1                hb9d3cd8_1    conda-forge
xorg-libxpm               3.5.17               hb9d3cd8_1    conda-forge
xorg-libxrandr            1.5.4                hb9d3cd8_0    conda-forge
xorg-libxrender           0.9.11               hb9d3cd8_1    conda-forge
xorg-libxt                1.3.0                hb9d3cd8_2    conda-forge
xorg-libxtst              1.2.5                hb9d3cd8_3    conda-forge
xorg-xorgproto            2024.1               hb9d3cd8_1    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zlib                      1.3.1                hb9d3cd8_2    conda-forge
zstandard                 0.23.0          py312hef9b889_1    conda-forge
zstd                      1.5.6                ha6fb4c9_0    conda-forge
```
**Command used and output returned**

```
$ poppunk_assign --db GPS_v4_references --query query_list.txt --output test --threads 12
PopPUNK: assign
	(with backend: sketchlib v2.1.4
	 sketchlib: /home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/pp_sketchlib.cpython-312-x86_64-linux-gnu.so)
Mode: Assigning clusters of query sequences


Graph-tools OpenMP parallelisation enabled: with 12 threads
Looking for existing sketches in test/test.h5
Loading previously refined model
Completed model loading
WARNING: versions of input databases sketches are different, results may not be compatible
Calculating distances using 12 thread(s)
Progress (CPU): 100.0%
Loading network from GPS_v4_references/GPS_v4_references_graph.gt
Network loaded: 2562 samples
Loading previous cluster assignments from GPS_v4_references/GPS_v4_references_clusters.csv
Found novel query clusters. Calculating distances between them.
Calculating all query-query distances
Using existing random match chances in DB
Calculating distances using 12 thread(s)
Progress (CPU): 100.0%
Clusters 1,5,8,9,10,13,14,21,25,26,28,35,36,37,38,42,45,47,48,49,50,56,60,64,67,69,71,73,75,76,77,80,102,111,130,132,140,145,146,152,155,175,178,179,185,193,197,200,206,209,219,227,229,230,235,242,243,252,256,261,264,272,273,292,293,307,309,321,322,326,334,340,342,353,360,366,397,413,425,439,463,471,483,511,512,515,518,540,551,556,559,561,591,593,595,631,637,649,686,689,694,701,705,712,717,720,721,722,730,732,737,742,755,774,776,777,786,799,824,830,835,836,837,849,858,863,888,891,900,906,912,914,926,934,939,957,984,990,992,1016,1017,1018,1030,1041,1053,1057,1061,1067,1086,1089,1093,1094 have merged into 1_5_8_9_10_13_14_21_25_26_28_35_36_37_38_42_45_47_48_49_50_56_60_64_67_69_71_73_75_76_77_80_102_111_130_132_140_145_146_152_155_175_178_179_185_193_197_200_206_209_219_227_229_230_235_242_243_252_256_261_264_272_273_292_293_307_309_321_322_326_334_340_342_353_360_366_397_413_425_439_463_471_483_511_512_515_518_540_551_556_559_561_591_593_595_631_637_649_686_689_694_701_705_712_717_720_721_722_730_732_737_742_755_774_776_777_786_799_824_830_835_836_837_849_858_863_888_891_900_906_912_914_926_934_939_957_984_990_992_1016_1017_1018_1030_1041_1053_1057_1061_1067_1086_1089_1093_1094
Traceback (most recent call last):
  File "/home/eriny/miniconda3/envs/poppunk/bin/poppunk_assign", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/PopPUNK/assign.py", line 213, in main
    assign_query(dbFuncs,
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/PopPUNK/assign.py", line 297, in assign_query
    isolateClustering = assign_query_hdf5(dbFuncs,
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/PopPUNK/assign.py", line 646, in assign_query_hdf5
    {'combined': printClusters(genomeNetwork,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/PopPUNK/network.py", line 1520, in printClusters
    unword = next(unword_generator)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/PopPUNK/unwords.py", line 31, in gen_unword
    word += "".join(syllable()[0]())
                    ^^^^^^^^^^^^^^^
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/PopPUNK/unwords.py", line 21, in <lambda>
    cvc = lambda: cv() + consonant()
                  ^^^^
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/PopPUNK/unwords.py", line 20, in <lambda>
    cv = lambda: consonant() + vowel()
                 ^^^^^^^^^^^
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/site-packages/PopPUNK/unwords.py", line 19, in <lambda>
    consonant = lambda: random.sample(consonants, 1)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/eriny/miniconda3/envs/poppunk/lib/python3.12/random.py", line 413, in sample
    raise TypeError("Population must be a sequence.  "
TypeError: Population must be a sequence.  For dicts or sets, use sorted(d).
```

**Describe the bug**
I would like to run poppunk for some S. pyogenes samples. I downloaded the database from the [url listed](https://ftp.ebi.ac.uk/pub/databases/pp_dbs/Streptococcus_pyogenes_v1_refs.tar.bz2) and extracted it.

I had 10 spades-assembled fasta sequences for some local S. pyogenes samples that we sequenced.

The fasta sequences look like this:
```
 head 2024CK-01722.fasta
>NODE_1_length_489079_cov_60.805069
AAGCTGCTATCGCAGCGGTTGATAAAGCGACAACAGCTGAAGGCATTAACCAAGAACTTG
GTAAAGGCATCACAGCTATCAATAAAGCTTACCGTCCAGGTGAAGGTGTTAAAGCACGTA
AAGAAGCCGCTAAGGCTGATCTTGAAAGAGAAGCTGCTAAGGTTCGTGAAGCTATCGCTA
ACGACCCAACCTTAACAAAAGCTGATAAAGCTAAACAAACAGAAGCTGTTGCTAAAGCTC
TTAAAGCTGCTATCGCAGCGGTTGATAAAGCGACAACAGCTGAAGGCATTAACCAAGAAC
TTGGTAAAGGCATCACAGCTATCAATAAAGCTTACCGTCCAGGTGAAGGTGTTAAAGCAC
GTAAAGAAGCTGCTAAAGCTAATCTTGAAAAAGTAGCTAAAGAAACTAAAGCTCTTATTT
CAGGAGACCGTTACTTGAGCGAAACTGAAAAAGCAGTCCAAAAACAAGCTGTTGAGCAAG
CTCTTGCGAAAGCACTTGGTCAAGTTGAGGCTGCTAAGACAGTTGAAGCTGTTAAGTTGG
```

My query list looks like this:
```
$ head query_list.txt 
2024CK-01731	2024CK-01731.fasta
2024CK-01723	2024CK-01723.fasta
2024CK-01730	2024CK-01730.fasta
2024CK-01722	2024CK-01722.fasta
2024CK-01734	2024CK-01734.fasta
2024CK-01732	2024CK-01732.fasta
2024CK-01737	2024CK-01737.fasta
2024CK-01733	2024CK-01733.fasta
2024CK-01735	2024CK-01735.fasta
2024CK-01736	2024CK-01736.fasta
```

I feel like I am doing something wrong, and would like some guidance.

**Comment by johnlees** [comment]
This is a duplicate of #321 i.e. a bug in v2.7.0 -- see the suggested fix over at this issue.

 Or you can also clone the current version of the repo and run the code, as it's fixed there (not released yet due to problems with bioconda)

**Comment by johnlees** [comment]
Fixed in v2.7.1

**Comment by erinyoung** [comment]
Thank you!
